### PR TITLE
Use time.Duration for timeout

### PR DESCRIPTION
**Pull request by kajinamit**
instead of int. This allows us to use more fine grained timeout value such as 0.1, and encodes the unit in the type itself.

Depends-On: https://github.com/openstack-k8s-operators/lib-common/pull/155

### CODE REVIEW

**Comment by kajinamit** [comment]
Still pending on https://github.com/openstack-k8s-operators/lib-common/pull/155

**Comment by kajinamit** [comment]
We probably have to merge this first, because the other operators depend on both lib-common and keystone-operator, it seems.

File: controllers/keystoneapi_controller.go [hunk 1]
```diff
@@ -291,7 +291,7 @@ func (r *KeystoneAPIReconciler) reconcileInit(
 		jobDef,
 		keystonev1.DbSyncHash,
 		instance.Spec.PreserveJobs,
-		5,
+		time.Duration(5)*time.Second,
```
Comment (gibizer) [inline controllers/keystoneapi_controller.go:294]: there is a job.NewJob call at L385 as well

Comment (gibizer) [reply controllers/keystoneapi_controller.go:294]: and there is a NewDeployment call at L573

Comment (gibizer) [reply controllers/keystoneapi_controller.go:294]: there is GetDataFromSecret calls in L84 here and in keystoneservice_controller L431

**Review by gibizer — CHANGES_REQUESTED** [review]

File: controllers/keystoneapi_controller.go (hunk 1 above)
Comment (kajinamit) [reply controllers/keystoneapi_controller.go:294]: Fixed these. also there was GetDataFromSecret in `api/v1beta1/keystoneapi.go` I had to fix...

**Review by gibizer — CHANGES_REQUESTED** [review]
Sorry but found more.
NewService and NewRoute calls in keystoneapi

**Comment by kajinamit** [comment]
> Sorry but found more. NewService and NewRoute calls in keystoneapi

Hmm. I can't find these. Are these in the latest master ?

**Comment by gibizer** [comment]
> > Sorry but found more. NewService and NewRoute calls in keystoneapi
> 
> Hmm. I can't find these. Are these in the latest master ?

Ohh sorry these are in lib-common. As https://github.com/openstack-k8s-operators/lib-common/pull/166 is merged we need to adapt https://github.com/openstack-k8s-operators/keystone-operator/blob/d6671c8d0e22e4b6d4b4350223d54554a2980775/controllers/keystoneapi_controller.go#L341-L347

**Comment by gibizer** [comment]
> > > Sorry but found more. NewService and NewRoute calls in keystoneapi
> > 
> > 
> > Hmm. I can't find these. Are these in the latest master ?
> 
> Ohh sorry these are in lib-common. As [openstack-k8s-operators/lib-common#166](https://github.com/openstack-k8s-operators/lib-common/pull/166) is merged we need to adapt
> 
> https://github.com/openstack-k8s-operators/keystone-operator/blob/d6671c8d0e22e4b6d4b4350223d54554a2980775/controllers/keystoneapi_controller.go#L341-L347

Done

**Review by abays — APPROVED** [review]
/lgtm

**Review by gibizer — APPROVED** [review]

**Pull request merged**